core/src/main/java/org/infinispan/security/actions/RemoveListenerAction.java

```diff
@@ -0,0 +1,29 @@
+package org.infinispan.security.actions;
+
+import java.security.PrivilegedAction;
+
+import org.infinispan.notifications.Listenable;
+
+/**
+ * RemoveListenerAction
+ *
+ * @author vjuranek
+ * @since 9.0
+ */
+public class RemoveListenerAction implements PrivilegedAction<Void> {
+
+    private final Listenable listenable;
+    private final Object listener;
+
+    public RemoveListenerAction(Listenable listenable, Object listener) {
+        this.listenable = listenable;
+        this.listener = listener;
+    }
+
+    @Override
+    public Void run() {
+        listenable.removeListener(listener);
```

Comment:
instead of doing nothing when `listenable` is `null`, we should at least log it, as there's probably something wrong, `null` is definitely expected here. Looking into other security actions, non of them contains logging or other checks - I guess it's because of performance, as these actions can be used quite frequently - is this correct @tristantarrant ?

If I'm wrong and there are no performance concerns, no problem to add it.